watertap/unit_models/nanofiltration_DSPMDE_0D.py

```diff
@@ -0,0 +1,1394 @@
+                var_dict[f"Osmotic Pressure @ Bulk Feed, Outlet (Pa)"] = \
+                    self.feed_side.properties_out[time_point].pressure_osm
+
+            for x in self.io_list:
+                if not x:
+                    io = 'Inlet'
+                    prop_feed = self.feed_side.properties_in[0]
+                elif x:
+                    io = 'Outlet'
+                    prop_feed = self.feed_side.properties_out[0]
+
+                var_dict[f'Molar Concentration of {j} @ Membrane Interface, {io}'] = \
+                    self.feed_side.properties_interface[time_point, x].conc_mol_phase_comp['Liq', j]
+                var_dict[f'Molar Concentration of {j} @ Pore Entrance, {io}'] = \
+                    self.pore_entrance[time_point, x].conc_mol_phase_comp['Liq', j]
+                var_dict[f'Molar Concentration of {j} @ Pore Exit, {io}'] = \
+                    self.pore_exit[time_point, x].conc_mol_phase_comp['Liq', j]
+                var_dict[f'Molar Concentration of {j} @ Permeate, {io}'] = \
+                    self.permeate_side[time_point, x].conc_mol_phase_comp['Liq', j]
+
+
+                var_dict[f"Osmotic Pressure @ Membrane Interface, {io} (Pa)"] = \
+                    self.feed_side.properties_interface[time_point, x].pressure_osm
+
+                var_dict[f"Osmotic Pressure @ Permeate, {io} (Pa)"] = \
+                    self.permeate_side[time_point, x].pressure_osm
+                expr_dict[f"Net Driving Pressure, {io} (Pa)"] = (prop_feed.pressure
+                                                                 - self.permeate_side[0, x].pressure
+                                                                 - (self.feed_side.properties_interface[0, x].pressure_osm
+                                                                    - self.permeate_side[0, x].pressure_osm))
+                var_dict[f'Electric Potential @ Pore Entrance, {io}'] = self.electric_potential[0, x, 'pore_entrance']
+                var_dict[f'Electric Potential @ Pore Exit, {io}'] = self.electric_potential[0, x, 'pore_exit']
+                var_dict[f'Electric Potential @ Permeate, {io}'] = self.electric_potential[0, x, 'permeate']
+                var_dict[f'Electric Potential Gradient @ Feed-Membrane Interface, {io}'] = \
+                    self.electric_potential_grad_feed_interface[0, x]
+
+        return {"vars": var_dict, "exprs": expr_dict}
+
+    def _get_stream_table_contents(self, time_point=0):
+        return create_stream_table_dataframe(
+            {
+                "Feed Inlet": self.inlet,
+                "Feed Outlet": self.retentate,
+                "Permeate Outlet": self.permeate,
+            },
+            time_point=time_point,
+        )
+
+    def get_costing(self, module=None, **kwargs):
+        self.costing = Block()
+        module.Nanofiltration_costing(self.costing, **kwargs)
+
+    def _get_state_args(self, source, mixed_permeate_properties, initialize_guess, state_args):
```

Comment:
Could you comment on the purpose of this method (it probably should be included in the doc string).

Reply:
I added something to describe this method. I adopted this from what @bknueven did for RO; it returns `state_args` for each of the stateblocks in the unit model which subsequently get passed to those stateblocks during initialization.

Reply:
Maybe we should ultimately put this method in a base class?